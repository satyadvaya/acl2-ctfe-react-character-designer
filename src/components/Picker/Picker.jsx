import React from "react";

export default function Picker({
  head,
  onHeadChange,
  middle,
  onMiddleChange,
  pants,
  onPantsChange,
  newPhrase,
  setNewPhrase,
  handleClick,
}) {
  const headImages = ["bird", "dog", "duck", "horse"];
  const middleImages = ["blue", "fancy", "pink", "red"];
  const pantsImages = ["blue", "dog", "leg", "white"];

  return (
    <div>
      <label>
        Head
        <select
          value={head}
          onChange={(event) => onHeadChange(event.target.value)}
        >
          {headImages.map((image) => {
            return <option key={image}>{image}</option>;
          })}
        </select>
      </label>

      <label>
        Middle
        <select
          value={middle}
          onChange={(event) => onMiddleChange(event.target.value)}
        >
          {middleImages.map((image) => {
            return <option key={image}>{image}</option>;
          })}
        </select>
      </label>

      <label>
        Pants
        <select
          value={pants}
          onChange={(event) => onPantsChange(event.target.value)}
        >
          {pantsImages.map((image) => {
            return <option key={image}>{image}</option>;
          })}
        </select>
      </label>

      <label>
        New Phrase
        <input
          type="text"
          value={newPhrase}
          onChange={(event) => setNewPhrase(event.target.value)}
        />
      </label>
      <button type="button" onClick={handleClick}>
        Submit Phrase!
      </button>
    </div>
  );
}
