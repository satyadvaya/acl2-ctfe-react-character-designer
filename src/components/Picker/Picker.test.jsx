import { screen, render } from "@testing-library/react";
import Picker from "./Picker";

it("should render the Picker component", () => {
  const { container } = render(<Picker />);

  const selectHead = screen.getByLabelText(/Head/);
  const selectMiddle = screen.getByLabelText(/Middle/);
  const selectPants = screen.getByLabelText(/Pants/);
  const selectPhrase = screen.getByLabelText(/New Phrase:/);

  expect(selectHead).toBeInTheDocument();
  expect(selectMiddle).toBeInTheDocument();
  expect(selectPants).toBeInTheDocument();
  expect(selectPhrase).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
