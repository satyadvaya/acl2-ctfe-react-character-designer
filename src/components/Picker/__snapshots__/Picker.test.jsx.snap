// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Picker component 1`] = `
<div>
  <div>
    <label>
      Head
      <select>
        <option>
          bird
        </option>
        <option>
          dog
        </option>
        <option>
          duck
        </option>
        <option>
          horse
        </option>
      </select>
    </label>
    <label>
      Middle
      <select>
        <option>
          blue
        </option>
        <option>
          fancy
        </option>
        <option>
          pink
        </option>
        <option>
          red
        </option>
      </select>
    </label>
    <label>
      Pants
      <select>
        <option>
          blue
        </option>
        <option>
          dog
        </option>
        <option>
          leg
        </option>
        <option>
          white
        </option>
      </select>
    </label>
    <label>
      New Phrase:
      <input
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Submit Phrase!
    </button>
  </div>
</div>
`;
